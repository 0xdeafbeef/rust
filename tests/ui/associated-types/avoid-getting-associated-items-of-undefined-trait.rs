// Fix for <https://github.com/rust-lang/rust/issues/137508>.

trait Tr {
    type Item;
}

fn main() {
    let _: dyn Tr + ?Foo<Assoc = ()>;
    //~^ ERROR: cannot find trait `Foo` in this scope
    //~| ERROR: `?Trait` is not permitted in trait object types
}
