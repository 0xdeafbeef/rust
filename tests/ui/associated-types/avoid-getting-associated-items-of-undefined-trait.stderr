error[E0405]: cannot find trait `Foo` in this scope
  --> $DIR/avoid-getting-associated-items-of-undefined-trait.rs:8:22
   |
LL |     let _: dyn Tr + ?Foo<Assoc = ()>;
   |                      ^^^ not found in this scope

error[E0658]: `?Trait` is not permitted in trait object types
  --> $DIR/avoid-getting-associated-items-of-undefined-trait.rs:8:21
   |
LL |     let _: dyn Tr + ?Foo<Assoc = ()>;
   |                     ^^^^^^^^^^^^^^^^
   |
   = help: add `#![feature(more_maybe_bounds)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0405, E0658.
For more information about an error, try `rustc --explain E0405`.
